components/CharacterFilter.tsx

```diff
@@ -23,75 +24,107 @@ type CharacterFaction = {
   imageUrl: string;
 };
 
-export default function CharacterFilter(props: {
+type Props = {
   characterFilter: CharacterFilterType;
-  updateCharacterFilter: (filter: CharacterFilterType) => void;
+  updateCharacterFilter(filter: CharacterFilterType): void;
   className?: string;
   classes: { [key: string]: CharacterClass };
   factions: { [key: string]: CharacterFaction };
-}) {
+};
+
+export default function CharacterFilter({
+  characterFilter,
+  updateCharacterFilter,
+  className,
+  classes,
+  factions,
+}: Props) {
+  const activeFactionFilter =
+    characterFilter.faction !== "All" &&
+    Object.entries(factions).find(
```

Comment:
why not just do `factions[characterFilter.faction.toLowerCase()]`

Reply:
Good shout in my head it was an Array so i guess i got tunnel visioned ahha